src/blueprints/settings/__init__.py

```
@@ -352,24 +372,97 @@ def _set_update_state(running: bool, unit: str | None):
+
+
+def _validate_update_script_path(script_path: str) -> str:
+    """Return a sanitised absolute script path or raise ``ValueError``.
+
+    The basename must be in :data:`_ALLOWED_UPDATE_SCRIPT_BASENAMES` and the
+    full path must match a strict regex of safe characters and end in an
+    allow-listed basename. CodeQL recognises ``re.fullmatch`` against a
+    safe-character class as a sanitiser for ``py/command-line-injection``.
+    """
+    if not isinstance(script_path, str) or not script_path:
+        raise ValueError(f"Invalid update script path: {script_path!r}")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)